packages/core/src/components/time-picker/time-picker.tsx

```diff
@@ -557,22 +567,25 @@ export class TimePicker {
   private mutationObserverCallback(mutations: MutationRecord[]) {
     for (const mutation of mutations) {
       if (mutation.type !== 'attributes') {
-        this.hasUpdatedScrollPositions = false;
         continue;
       }
 
       const dropdown = mutation.target as HTMLElement;
+
       if (!dropdown.classList.contains('show')) {
+        // keep picker in sync with input
+        this._time = DateTime.fromFormat(this.time, this.format);
```

Comment:
Although we check for absence of .show here the dropdown will still update while open, which is not desired AFAIK.